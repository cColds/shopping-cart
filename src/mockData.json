{
  "data": {
    "products": {
      "edges": [
        {
          "node": {
            "id": "gid://shopify/Product/7982853619734",
            "title": "Slides",
            "description": "Simple, minimal and comfortable, these slides feature a classic design in the perfect shade of iron. Whether you're just lounging around the house or running errands, these slides will offer all-day comfort.",
            "featuredImage": {
              "id": "gid://shopify/ProductImage/39774608687126",
              "url": "https://cdn.shopify.com/s/files/1/0688/1755/1382/products/slides.jpg?v=1675447358"
            },
            "variants": {
              "edges": [
                {
                  "node": {
                    "price": {
                      "amount": "25.0",
                      "currencyCode": "CAD"
                    }
                  }
                }
              ]
            }
          }
        },
        {
          "node": {
            "id": "gid://shopify/Product/7982856273942",
            "title": "Sweatpants",
            "description": "Soft and comfortable sweatpants in stylish shades. They are perfect for lounging with their cozy stretch fabric that offers just the right amount of warmth. Enjoy the ultimate relaxation experience!",
            "featuredImage": {
              "id": "gid://shopify/ProductImage/39774602002454",
              "url": "https://cdn.shopify.com/s/files/1/0688/1755/1382/products/GreenSweatpants01.jpg?v=1675455387"
            },
            "variants": {
              "edges": [
                {
                  "node": {
                    "price": {
                      "amount": "35.0",
                      "currencyCode": "CAD"
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}
