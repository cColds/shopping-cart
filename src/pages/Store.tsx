import ItemCards from "../components/ItemCards";
import Item from "../interfaces/Item";

export default function Store({ items }: { items: Item[] }) {
  return (
    <section className="flex w-full flex-col items-center p-8">
      <ItemCards items={items} />
    </section>
  );
}
