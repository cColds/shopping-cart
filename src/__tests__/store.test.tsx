import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Store from "../pages/Store";
import mockData from "../mockData.json";

// test that items show up when going to store

it("should fetch mock.shop api", async () => {
  (fetch as any).mockResponseOnce(JSON.stringify(mockData));

  render(
    <MemoryRouter>
      <Store />
    </MemoryRouter>
  );
  expect(await screen.findByText(/slides/i)).toBeInTheDocument();
  expect(await screen.findByText(/sweatpants/i)).toBeInTheDocument();
});

it.skip("should fail to fetch mock.shop api", async () => {
  (fetch as any).mockRejectOnce(new Error("Failed to fetch!"));
  render(
    <MemoryRouter>
      <Store />
    </MemoryRouter>
  );
  expect(screen.queryByText(/slides/i)).not.toBeInTheDocument();
  expect(screen.queryByText(/sweatpants/i)).not.toBeInTheDocument();
});
