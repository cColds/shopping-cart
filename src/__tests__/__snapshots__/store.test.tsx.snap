// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render store and match snapshot 1`] = `
<div>
  <section
    class="flex w-full flex-col items-center p-8"
  >
    <ul
      class="m-8 flex w-full flex-wrap justify-center gap-12 p-8"
    >
      <li
        class="flex cursor-pointer flex-col items-center transition duration-150 ease-in-out active:scale-95"
      >
        <a
          aria-label="Slides for $25"
          class="w-[200px]"
          href="/store/slides"
        >
          <img
            alt="Slides"
            class="rounded-tl-lg rounded-tr-lg"
            draggable="false"
            height="200px"
            src="https://cdn.shopify.com/s/files/1/0688/1755/1382/products/slides.jpg?v=1675447358"
            width="200px"
          />
          <div
            class="flex h-24 flex-col gap-1 rounded-bl-lg rounded-br-lg bg-white p-2.5 shadow-lg"
          >
            <h3
              class="truncate"
            >
              Slides
            </h3>
            <p
              class="truncate"
            >
              $
              25
            </p>
          </div>
        </a>
      </li>
      <li
        class="flex cursor-pointer flex-col items-center transition duration-150 ease-in-out active:scale-95"
      >
        <a
          aria-label="Sweatpants for $35"
          class="w-[200px]"
          href="/store/sweatpants"
        >
          <img
            alt="Sweatpants"
            class="rounded-tl-lg rounded-tr-lg"
            draggable="false"
            height="200px"
            src="https://cdn.shopify.com/s/files/1/0688/1755/1382/products/GreenSweatpants01.jpg?v=1675455387"
            width="200px"
          />
          <div
            class="flex h-24 flex-col gap-1 rounded-bl-lg rounded-br-lg bg-white p-2.5 shadow-lg"
          >
            <h3
              class="truncate"
            >
              Sweatpants
            </h3>
            <p
              class="truncate"
            >
              $
              35
            </p>
          </div>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
